src/stack.jl

```diff
@@ -135,14 +135,31 @@ function gen_bidirectional_stack{TL<:HasLengthUnits, TT<:HasTimeUnits, TTI<:HasI
     #offset by one sample going forward so that we don't use the end points of the triangle
     delay1samp = 1/sample_rate
 
-    exp_intervals_fwd = spaced_intervals(posfwd, z_spacing, exp_time, sample_rate; delay=delay1samp, z_pad = z_pad, alignment=:start, rig=rig)
-    exp_intervals_back = spaced_intervals(posback, z_spacing, exp_time, sample_rate; delay=0.0*Unitful.s, z_pad = z_pad, alignment=:stop, rig=rig)
+    #We want the cameras to reach the global exposure state at the same time for each pulse in forward and reverse stacks, so we
+    #don't line up the pulses exactly.  Assuming that the exposure is one las_frac's duration longer than the minimum exposure time for the given
+    #should really put some more work into making this modular and adaptable to different exposure times / ROI sizes / line times /cameras
+    #For now we deliver the flash at the end of the exposure pulse, and we align the forward and reverse stacks to the flash (not to the exposure)
+    #Therefor it's up to the user to make sure the exposure time and ROI size combination is long enough that the flash fits within the global
+    #exposure period (see PCO.Edge manual for more details)
+
+    #distance traveled in one exposure time.  We need to adjust padding by that much to make sure that exposures can fit.
+    exp_dist = exp_time * abs(pmax - pmin) / stack_time
+    if exp_dist > z_pad
+        warn("Increasing z padding by $(exp_dist-z_pad) to leave space for bidirectional exposures\n")
+    end
+    exp_intervals_fwd = spaced_intervals(posfwd, z_spacing, exp_time, sample_rate; delay=delay1samp, z_pad = max(z_pad,exp_dist), alignment=:start, rig=rig)
+    flash_nsamps = calc_num_samps(min(flash_frac,1.0) * exp_time, sample_rate)
+    exp_nsamps = width(exp_intervals_fwd[1]) + 1
+    offset_nsamps = exp_nsamps - flash_nsamps
+    #The extra -1 is needed since the forward direction does not sample the last point (the first index in the reverse direction)
+    exp_intervals_back = map(x-> ClosedInterval(length(posfwd)-(maximum(x)+offset_nsamps-2), length(posfwd)-(maximum(x)-flash_nsamps-1)), exp_intervals_fwd)
```

Comment:
You may already be aware of this, but a shortcut for `ClosedInterval(a, b)` is `a..b`. In cases where the arguments are this long, however, it might actually be clearer to read this way.